src/radiation/geodesics.hpp

```diff
@@ -35,17 +35,48 @@ void GetXSource(Real &Kcon0, Real &Kcon1, Real &Kcon2, Real &Kcon3, Real src[NDF
   src[3] = Kcon3 / Kcon0;
 }
 
-KOKKOS_INLINE_FUNCTION
-void GetKSource(Real &X0, Real &X1, Real &X2, Real &X3, Real &Kcov0, Real &Kcov1,
-                Real &Kcov2, Real &Kcov3, Real &Kcon0,
-                const Geometry::CoordSysMeshBlock &geom, Real source[4]) {
-  SPACETIMELOOP(mu) { source[mu] = 0.; }
+template <class Geom>
+KOKKOS_INLINE_FUNCTION void GetKSource(Real &X0, Real &X1, Real &X2, Real &X3,
+                                       Real &Kcov0, Real &Kcov1, Real &Kcov2, Real &Kcov3,
+                                       Real &Kcon0, Geom &geom, Real source[4]) {
```

Comment:
We should probably add a test to check this is correct.

Reply:
Photon orbits or something. That can be for later though. Not this MR.

Reply:
Agreed. #226